arithmetic_analysis/newton_raphson_method.py

```diff
@@ -4,31 +4,45 @@
 # quickly find a good approximation for the root of a real-valued function
 from sympy import diff
 from decimal import Decimal
+from math import *
+
+
+PRECISION = 10 ** -10
 
 
 def NewtonRaphson(func, a):
-    """ Finds root from the point 'a' onwards by Newton-Raphson method """
+    """ Finds root from the point 'a' onwards by Newton-Raphson method
+    >>> NewtonRaphson("sin(x)", 2))
+    3.141592653680804119362982272
+    >>> NewtonRaphson("x**2 - 5*x +2", 0.4))
+    0.4384471871911695062638833812
+    >>> NewtonRaphson("x**2 - 5", 0.1))
+    2.236067977499789696465261251
+    >>> NewtonRaphson("log(x)- 1", 2))
+    2.718281828458937968804105873
+    """
+    
+    x = a
     while True:
-        c = Decimal(a) - (Decimal(eval(func)) / Decimal(eval(str(diff(func)))))
 
-        a = c
+        x = Decimal(x) - (Decimal(eval(func)) / Decimal(eval(str(diff(func)))))
```

Comment:
`Decimal` class is present in `decimal` module, import that if you want to use it.

[Source](https://docs.python.org/3.8/library/decimal.html#decimal.Decimal)